File: ITP_KANDY_7-main/backend/src/domain/entities/Supplier.js
/**
 * Domain Layer: Supplier Entity.
 * Represents a business partner providing goods to the merchant.
 * Encapsulates contact details and the current balance of accounts payable.
 */
class Supplier {
    /**
     * Logic: Entity Initialization.
     * Maps partner metadata into a structured business object for supply chain management.
     */
    constructor({ 
        id, // Unique identity (UUID)
        name, // Official business or person name
        phone, // Primary contact number
        address, // Physical location for deliveries
        email, // Digital contact address
        notes, // Optional merchant annotations
        status, // State: 'active' or 'inactive'
        totalPayable, // Aggregate debt: The total amount currently owed to this supplier
        createdAt, // Audit: Genesis date
        updatedAt // Audit: Last modification timestamp
    }) {
        this.id = id;
        this.name = name || '';
        this.phone = phone || '';
        this.address = address || '';
        this.email = email || '';
        this.notes = notes || '';
        this.status = status || 'active';
        this.totalPayable = totalPayable || 0; // Logic: Incremented on purchases, decremented on payments
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    /**
     * Logic: Data Serialization.
     * Converts the entity into a JSON object for API transmission to the Flutter frontend.
     */
    toJSON() {
        return {
            id: this.id,
            name: this.name,
            phone: this.phone,
            address: this.address,
            email: this.email,
            notes: this.notes,
            status: this.status,
            totalPayable: this.totalPayable, // Critical: Used for dashboard debt totals
            createdAt: this.createdAt,
            updatedAt: this.updatedAt,
        };
    }
}

// Module Export: Business logic representation of a supply partner.
module.exports = Supplier;

File: ITP_KANDY_7-main/frontend/lib/features/suppliers/domain/entities/supplier.dart
// ------------------------------------------------------------------------------
// File: supplier.dart
// Purpose: Domain entity representing an external business partner/supplier.
// Rationale: Provides a strongly-typed model for supplier profile data
//   including aggregate payable balance. Used across Supplier module screens,
//   providers, and PDF export utilities.
// ------------------------------------------------------------------------------
class Supplier {
  final String id;
  final String name;
  final String phone;
  final String address;
  final String email;
  final String notes;
  final String status;
  final double totalPayable;
  final String createdAt;

  Supplier({
    required this.id,
    required this.name,
    this.phone = '',
    this.address = '',
    this.email = '',
    this.notes = '',
    this.status = 'active',
    this.totalPayable = 0, // The amount of money the shop owner owes this supplier
    this.createdAt = '',
  });

  factory Supplier.fromJson(Map<String, dynamic> json) {
    return Supplier(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      address: json['address'] ?? '',
      email: json['email'] ?? '',
      notes: json['notes'] ?? '',
      status: json['status'] ?? 'active',
      totalPayable: (json['totalPayable'] ?? 0).toDouble(),
      createdAt: json['createdAt'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
      'address': address,
      'email': email,
      'notes': notes,
    };
  }
}

